applications/FluidDynamicsApplication/custom_utilities/drag_utilities.cpp

```diff
@@ -33,67 +33,68 @@ namespace Kratos
+            unsigned int solution_step = r_current_process_info[STEP];
+            if (solution_step+1 >= rModelPart.GetBufferSize() ) {
 
-            #pragma omp parallel for private(RHS_Contribution, LHS_Contribution) firstprivate(domain_size)
-            for (int i_elem = 0; i_elem < static_cast<int>(root_model_part.NumberOfElements()); ++i_elem){
-                auto it_elem = root_model_part.ElementsBegin() + i_elem;
+                #pragma omp parallel for private(RHS_Contribution, LHS_Contribution) firstprivate(domain_size)
+                for (int i_elem = 0; i_elem < static_cast<int>(root_model_part.NumberOfElements()); ++i_elem){
+                    auto it_elem = root_model_part.ElementsBegin() + i_elem;
 
-                // Build local system
-                it_elem->CalculateLocalSystem(LHS_Contribution, RHS_Contribution, r_current_process_info);
+                    // Build local system
+                    it_elem->CalculateLocalSystem(LHS_Contribution, RHS_Contribution, r_current_process_info);
 
-                // Get geometry
-                Element::GeometryType& r_geom = it_elem->GetGeometry();
-                const unsigned int n_nodes = r_geom.PointsNumber();
-                const unsigned int block_size = RHS_Contribution.size()/n_nodes;
+                    // Get geometry
+                    Element::GeometryType& r_geom = it_elem->GetGeometry();
+                    const unsigned int n_nodes = r_geom.PointsNumber();
+                    const unsigned int block_size = RHS_Contribution.size()/n_nodes;
 
-                for (unsigned int i_node = 0; i_node < n_nodes; ++i_node){
-                    r_geom[i_node].SetLock();
-                    array_1d<double,3>& r_reaction = r_geom[i_node].FastGetSolutionStepValue(REACTION);
-                    for (unsigned int d = 0; d < domain_size; ++d)
-                        r_reaction[d] -= RHS_Contribution[block_size*i_node + d];
+                    for (unsigned int i_node = 0; i_node < n_nodes; ++i_node){
+                        r_geom[i_node].SetLock();
+                        array_1d<double,3>& r_reaction = r_geom[i_node].FastGetSolutionStepValue(REACTION);
+                        for (unsigned int d = 0; d < domain_size; ++d)
+                            r_reaction[d] -= RHS_Contribution[block_size*i_node + d];
 
-                    r_geom[i_node].UnSetLock();
+                        r_geom[i_node].UnSetLock();
+                    }
                 }
-            }
+                
+                // Restore the step index
+                r_current_process_info[FRACTIONAL_STEP] = current_fractional_step;
 
-            // Assemble reaction data
-            root_model_part.GetCommunicator().AssembleCurrentData(REACTION);
+                // Assemble reaction data
+                root_model_part.GetCommunicator().AssembleCurrentData(REACTION);
+            }
         }
```

Comment:
To be honest I'm not convinced this is required, even for the case of fractional step (you are doing the same thing when you call FSStrategy.CalculateReactions() in the python solver.

Reply:
If it is already done in the FS strategy we can remove it. Isn't it?

Reply:
I'd say yes, but I'd recommend a small test to verify. Also, we need to make sure that the reactions flag is checked whenever we enable the drag utility (this is also true for the monolithic case).

Reply:
Totally agree about checking the flag. The point is that this needs to be done at the interface level since the drag process doesn't know anything about the strategy....

Regarding the communicator issue, let me do a simple MPI test.

Reply:
If you go for MPI, please check if there is the same indentation issue in the trilinos solver.

Reply:
No difference appears if we remove the Assemble we are discussing.  Regarding the indentation issue, the Trilinos solver is taking the solve method from the base one, meaning that there is no problem.

Reply:
OK, I'm really lost now... What I meant to say is that the whole calculating the reactions deal was not required if you are already calculating the reactions using the FS strategy, so maybe you also don't need to do the calculation again in the fractional step case (so you never calculate in the body-fitted solvers, just rely on the builder and solver and add the existing reactions).

The assemble itself should be needed in the MPI case, but it is only noticeable in the domain boundaries. Maybe your drag surface did not cross through an MPI boundary?